lib/web_ui/test/mobile_unit_test_bindings.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:html' as html;
+import 'dart:js';
+import 'dart:js_util' as js_util;
+
+class MobileUnitTestBindings {
+  final Completer<bool> _allTestsPassed = Completer<bool>();
+
+  void tearDown() {
+    if (!_allTestsPassed.isCompleted) {
+      _allTestsPassed.complete(true);
+    }
+  }
+
+  void initialize() {
+    Future<Map<String, dynamic>> callback(Map<String, String> params) async {
+      final String command = params['command'];
+      Map<String, String> response;
+      switch (command) {
+        case 'fetch_test_results':
+          final bool allTestsPassedFlag = await _allTestsPassed.future;
+          response = <String, String>{
+            // TODO(nurhan): Send a more detailed error message. We can
```

Comment:
I am planning to use the same improvement I did for e2e, and use a Response class which has the stack trace: https://github.com/flutter/plugins/blob/master/packages/e2e/lib/common.dart#L9

Let me know if you have concerns.